src/care/host_device_ptr.h

```diff
@@ -396,6 +398,16 @@ namespace care {
       }
    }; // class host_device_ptr
 
+
+   // Prints host_device data to the given ostream. 
+   // When ENABLE_IMPLICIT_CONVERSIONS is off, this line is needed for the line 'using stream_insertion_t = decltype(std::cout << std::declval<T>());' in
```

Comment:
I'm impressed you figured this out in the midst of all those template shenanigans.

Reply:
Just use doxygen style comments - I have it set up so that we actually can generate documentation from the code.

Reply:
Added the comments in doxygen style, no problem. 

It took me a long while staring blankly at strange template error messages to figure that one out :D . But yes, I learned a bit, and Google helped a bit :D . Glad that is resolved!